File: Fore-C_v2/codes/Model_summarize_results_across_smote_datasets.R
# load libraries
library(ggplot2)
library(tidyverse)

# load data
x <- read.csv("../model_selection_summary_results/qf_smote_summary.csv")

# format data for plotting
x2 <- x %>%
  filter(selection == "parsimonious_best") %>%
  group_by(Disease_type) %>%
  filter(AdjR2_withheld_sample == max(AdjR2_withheld_sample))

# sort(strsplit(x2$Covariates[4], ", ")[[1]])

# save
write.csv(x2, 
          "../model_selection_summary_results/parsimonious_best_models_by_disease_and_region.csv", 
          row.names = F)

File: Fore-C_v2/codes/Model_summarize_results_across_smote_datasets.py
# -*- coding: utf-8 -*-
# load libraries
import pandas as pd

# load data
x = pd.read_csv("../model_selection_summary_results/qf_smote_summary.csv")

# select rows with best models
# select rows based on parsimonious best models
x2 = x.loc[(x['selection'] == 'parsimonious_best')]

# for each disease-region, select rows with highest Adj R2 values 
x2 = x2.sort_values('AdjR2_withheld_sample', ascending = False).drop_duplicates(['Disease_type'])

# save
x2.to_csv('../model_selection_summary_results/parsimonious_best_models_by_disease_and_region.csv', index = False)
